rackcity/views/it_instance_views.py

```diff
@@ -192,16 +244,53 @@ def instance_page_count(request):
     return JsonResponse({"page_count": page_count})
 
 
-def is_location_full(rack_id, instance_elevation, instance_height):
+def is_location_full(rack_id, instance_id, instance_elevation, instance_height):
     new_instance_location_range = [
         instance_elevation + i for i in range(instance_height)
     ]
     instances_in_rack = ITInstance.objects.filter(rack=rack_id)
     for instance_in_rack in instances_in_rack:
-        for occupied_location in [
-            instance_in_rack.elevation + i for i
-                in range(instance_in_rack.model.height)
-        ]:
-            if occupied_location in new_instance_location_range:
-                return True
+        # Ignore if instance being modified conflicts with its old location
+        if (instance_in_rack.id != instance_id):
+            for occupied_location in [
+                instance_in_rack.elevation + i for i
+                    in range(instance_in_rack.model.height)
+            ]:
+                if occupied_location in new_instance_location_range:
+                    return True
     return False
+
+
+def validate_location_modification(data, existing_instance):
```

Comment:
What would you think about making this function agnostic to whether it's a modification or an addition of an instance? For example by getting rid of the `existing_instance` parameter and then within the function checking if `'id' in data` to determine if there's an existing instance.

I think it would be generally nice to have the validation be the same for both cases, and it would be especially helpful for bulk import because additions and modifications happen all together and otherwise undergo the same validation logic.

I'm not really sure about this so feel free to dissent

Reply:
while I'm all about reducing duplicated code where we can, the difference between the two cases right now is that we use the serializer to validate incoming JSON data for `/instances/add` endpoint, and we can't use it for modifications because of the required field issue. I don't think this is worth getting rid of the serializer on adds because it's a more thorough validation. I think in the future we'll definitely want to move towards a more universal type of validation on each model, and I'm not sure what that'll look like yet, but I'm hesitant about throwing it all into this `validate_location_modification` method

thoughts???

Reply:
I meant making this method which is just for validating location agnostic to add vs modification, not necessarily the rest of the validation logic like for fields and uniqueness.  But if you disagree then I don't mind approving it